Clean global scope in controllers

galnitsan opened the pull request:
<!-- PLEASE READ THE FOLLOWING INSTRUCTIONS -->

### Proposed solution
1. functions moved inside controllers.

2.  suitable names were named for these functions. the "old" names of functions were changed to the "new" names in all functions calls.

3.  2 new services were create (factory) - (camps/suppliers). 

4. camps/suppliers services were injected to relevant controllers.

5. camps/suppliers services were declared in layout.jade (under "spark scripts").  

### Tradeoffs



### Testing Done
tested - ok.

<!-- BEFORE SUBMITTING YOUR PR, MAKE SURE TO FOLLOW THESE STEPS: -->
<!-- 1. Pull the latest `master` branch -->
<!-- 2. Run `npm install` to install all Spark dependencies -->
<!-- 3. Make sure your code is compliant with the [Spark styleguide](CONTRIBUTING.md) -->
<!-- 4. If your PR fixes an issue, reference that issue -->
<!-- 5. If your PR has lots of commits, **rebase** first -->
<!-- 6. Run `npm test` before submitting your PR -->

galnitsan committed:
clean global scope in controllers

galnitsan committed:
Update README.md

function moved inside controllers.
suitable names were named for these functions.
the "old" names of functions were changed to the "new" names in all functions calls.
2 new services were create (factory) - (camps/suppliers).
camps/suppliers services were injected to relevant controllers.
camps/suppliers services were declared in layout.jade (under "spark scripts").

galnitsan committed:
Update README.md

galnitsan committed:
Update README.md

@NitayRabi

galnitsan committed:
10-12-18

galnitsan committed:
Merge branch 'master' of https://github.com/Midburn/spark into Clean-global-scope-in-controllers

NitayRabi commented on README.md:
```diff
@@ -1,43 +1,9 @@
+camps/suppliers services were injected to relevant controllers.
+camps/suppliers services were declared in layout.jade (under "spark scripts").
+ still 1 issue in suppliers edit screen.
+ 
```
Copy this into the PR's description. this is the main project's readme

NitayRabi commented on package.json:
```diff
@@ -27,7 +27,7 @@
     "lint": "./node_modules/.bin/eslint .",
     "//createdb": "bootstraps MySql with a clean spark database & user",
-    "createdb": "mysql -u root < migrations/create_db.sql",
+    "createdb": "mysql -u root -p < migrations/create_db.sql",
```
revert

NitayRabi commented on public/scripts/controllers/camp_edit.js:
```diff
@@ -175,8 +26,158 @@ app.controller("campEditController", ($scope, $http, $filter, $q) => {
+    })
+
+    return deleteFilePromise.promise
+}
```
fix indentation

NitayRabi commented on public/scripts/controllers/camp_edit.js:
```diff
@@ -175,8 +26,158 @@ app.controller("campEditController", ($scope, $http, $filter, $q) => {
+//                 })
+//             }
+//         });
+// }
```
remove all this commented code

NitayRabi commented on public/scripts/controllers/camp_edit.js:
```diff
@@ -175,8 +26,158 @@ app.controller("campEditController", ($scope, $http, $filter, $q) => {
+    })
+
+    return getFilePromise.promise
+}
```
fix indentation

NitayRabi commented on public/scripts/controllers/camp_manage.js:
```diff
@@ -2,27 +2,27 @@
+//             on_success(res);
+//         });
+//     }
+// };
```
remove commented code

NitayRabi commented on public/scripts/services/camps_service.js:
```diff
@@ -0,0 +1,302 @@
+// angular_getCamps function from supplier_edit.js
+app.factory('camps', function () {
```
create a single function called campsFactory and then use it in both apps - 
```javascript
function campsFactory () {
    ....
    ....
}
app.factory('camps', campsFactory );
suppliers_app.factory('camps', campsFactory )
```

galnitsan committed:
fixed:
1. Identation
2.Removed commented lines
3.In camps_service.js - created single function "campsFactory" to use for  the factories (services) : app & suppliers.app

galnitsan committed:
fixed first line comments

galnitsan committed:
fixed issue of 2 global vars in camps_manage.js 1 deleted and 1 moved to camps_service.

NitayRabi committed:
Merge branch 'master' into Clean-global-scope-in-controllers
